silkworm/rpc/types/call.hpp

```diff
@@ -45,6 +45,10 @@ struct Call {
     std::optional<uint64_t> nonce;
     AccessList access_list;
 
+    void set_gas_price(const intx::uint256& new_gas_price) {
+        gas_price = new_gas_price;
+    }
+
```

Comment:
I would remove this and change the `to_transaction` signature to become:
```
[[nodiscard]] silkworm::Transaction to_transaction(std::optional<intx::uint256> override_gas_price) const
```
so that you can pass a value to override the `Call` gas price just when using it as a factory for a `Transaction` object.

The same reasoning also applies to `access_list`, I suggest doing the same change for that in a separate PR.

Reply:
ok